### PR TITLE
Use scipy.linalg.svd over numpy

### PR DESCRIPTION
**Pull request by tmct**
- [ ] closes #xxxx
- [ ] tests added / passed. 
- [ ] code/documentation is well formatted.  
- [ ] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

I've seen scipy's LAPACK wrapper give a significant performance increase compared to numpy, both in elapsed time and memory usage. It looks like others have come to this conclusion too: https://stackoverflow.com/a/58137088

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by josef-pkt** [comment]
That was a moving target.
When we started with this (10 years ago) numpy.linalg.pinv was much faster than scipy.linalg.
Many internals of scipy.linalg have changed.

can you do some timing for our common cases (nobs, k_vars) with nobs much larger than k_vars, e.g 1 Million by 100 or by 500

File: statsmodels/tools/tools.py [hunk 1]
```diff
@@ -506,7 +506,7 @@ def fullrank(x, r=None):
     if r is None:
         r = np.linalg.matrix_rank(x)
```
Comment (bashtage) [inline statsmodels/tools/tools.py:507]: This call is wasteful since it is performing a second svd.

Comment (josef-pkt) [reply statsmodels/tools/tools.py:507]: this function is not used, old code

We keep some of the old linalg functions as recipes, until we have our own toolset for linalg

**Comment by tmct** [comment]
Yes, definitely, I'll do some timing tests in a couple of days before declaring it ready - thanks for the suggestions.

**Comment by bashtage** [comment]
Looks like there may be a bad test somewhere.

**Comment by bashtage** [comment]
That SO post seems to be out-of-date.  I ran the notebook there and the NumPy and SciPy times are identical on both Windows and Linux, whether using OpenBLAS or MKL.

image

**Comment by bashtage** [comment]
Thanks for looking into this.

**Comment by tmct** [comment]
I have worked out why I was seeing different times for numpy - I was using numpy with MKL and scipy without MKL... Sorry about that!

No problem, will keep on the lookout for genuine improvements. Thanks for checking up on this.

**Pull request closed without merging**